Description: Fix the compatibility with Java 11:
 - No longer use the @Generated annotation which was removed
 - Add JAXB and JAF to the classpath
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: http://bugzilla.scilab.org/show_bug.cgi?id=15328

--- a/scilab/modules/graphic_objects/src/java/org/scilab/modules/graphic_objects/xmlloader/CSSParser.java
+++ b/scilab/modules/graphic_objects/src/java/org/scilab/modules/graphic_objects/xmlloader/CSSParser.java
@@ -14,7 +14,6 @@
 import java.util.HashMap;
 import java.util.Map;
 
-@javax.annotation.Generated("JFlex")
 @SuppressWarnings("fallthrough")
 
 /**
--- a/scilab/modules/helptools/src/java/jflex/c.jflex
+++ b/scilab/modules/helptools/src/java/jflex/c.jflex
@@ -6,7 +6,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/helptools/src/java/jflex/function.jflex
+++ b/scilab/modules/helptools/src/java/jflex/function.jflex
@@ -5,7 +5,6 @@
 import java.io.IOException;
 import java.io.StringReader;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/helptools/src/java/jflex/java.jflex
+++ b/scilab/modules/helptools/src/java/jflex/java.jflex
@@ -8,7 +8,6 @@
 
 import org.scilab.modules.helptools.c.AbstractCCodeHandler;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/helptools/src/java/jflex/scilab.jflex
+++ b/scilab/modules/helptools/src/java/jflex/scilab.jflex
@@ -12,7 +12,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/helptools/src/java/jflex/xml.jflex
+++ b/scilab/modules/helptools/src/java/jflex/xml.jflex
@@ -6,7 +6,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/helptools/src/java/org/scilab/modules/helptools/SynopsisLexer.java
+++ b/scilab/modules/helptools/src/java/org/scilab/modules/helptools/SynopsisLexer.java
@@ -9,8 +9,6 @@
 import java.io.IOException;
 import java.io.StringReader;
 
-@javax.annotation.Generated("JFlex")
-
 // See https://github.com/jflex-de/jflex/issues/222
 @SuppressWarnings("FallThrough")
 public final class SynopsisLexer {
--- a/scilab/modules/helptools/src/java/org/scilab/modules/helptools/XML/XMLLexer.java
+++ b/scilab/modules/helptools/src/java/org/scilab/modules/helptools/XML/XMLLexer.java
@@ -10,7 +10,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 
 // See https://github.com/jflex-de/jflex/issues/222
 @SuppressWarnings("FallThrough")
--- a/scilab/modules/helptools/src/java/org/scilab/modules/helptools/scilab/ScilabLexer.java
+++ b/scilab/modules/helptools/src/java/org/scilab/modules/helptools/scilab/ScilabLexer.java
@@ -16,7 +16,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 
 @SuppressWarnings("fallthrough")
 public final class ScilabLexer {
--- a/scilab/modules/scinotes/src/java/jflex/function.jflex
+++ b/scilab/modules/scinotes/src/java/jflex/function.jflex
@@ -8,7 +8,6 @@
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/scinotes/src/java/jflex/indentation.jflex
+++ b/scilab/modules/scinotes/src/java/jflex/indentation.jflex
@@ -6,7 +6,6 @@
 import javax.swing.text.Element;
 import javax.swing.text.BadLocationException;
 
-@javax.annotation.Generated("JFlex")
 %% 
 
 %public
--- a/scilab/modules/scinotes/src/java/jflex/matchingblock.jflex
+++ b/scilab/modules/scinotes/src/java/jflex/matchingblock.jflex
@@ -5,7 +5,6 @@
 import java.io.IOException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 %%
 
 %public
--- a/scilab/modules/scinotes/src/java/jflex/scilab.jflex
+++ b/scilab/modules/scinotes/src/java/jflex/scilab.jflex
@@ -12,7 +12,6 @@
 import javax.swing.text.Element;
 import org.scilab.modules.commons.ScilabCommonsUtils;
 
-@javax.annotation.Generated("JFlex")
 
 %%
 
--- a/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/IndentScanner.java
+++ b/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/IndentScanner.java
@@ -10,7 +10,6 @@
 import javax.swing.text.Element;
 import javax.swing.text.BadLocationException;
 
-@javax.annotation.Generated("JFlex")
 
 @SuppressWarnings("fallthrough")
 public final class IndentScanner {
--- a/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/MatchingBlockScanner.java
+++ b/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/MatchingBlockScanner.java
@@ -9,7 +9,6 @@
 import java.io.IOException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 
 @SuppressWarnings("fallthrough")
 public final class MatchingBlockScanner {
--- a/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/ScilabLexer.java
+++ b/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/ScilabLexer.java
@@ -16,7 +16,6 @@
 import javax.swing.text.Element;
 import org.scilab.modules.commons.ScilabCommonsUtils;
 
-@javax.annotation.Generated("JFlex")
 
 
 @SuppressWarnings("fallthrough")
--- a/scilab/modules/helptools/src/java/org/scilab/modules/helptools/c/CLexer.java
+++ b/scilab/modules/helptools/src/java/org/scilab/modules/helptools/c/CLexer.java
@@ -10,7 +10,6 @@
 import java.io.StringReader;
 import java.io.IOException;
 
-@javax.annotation.Generated("JFlex")
 
 // See https://github.com/jflex-de/jflex/issues/222
 @SuppressWarnings("FallThrough")
--- a/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/FunctionScanner.java
+++ b/scilab/modules/scinotes/src/java/org/scilab/modules/scinotes/FunctionScanner.java
@@ -12,7 +12,6 @@
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Element;
 
-@javax.annotation.Generated("JFlex")
 
 @SuppressWarnings("fallthrough")
 public final class FunctionScanner {
--- a/scilab/build.incl.xml
+++ b/scilab/build.incl.xml
@@ -58,6 +58,8 @@
         <pathelement location="${javafx.base.jar}"/>
         <pathelement location="${javafx.swing.jar}"/>
         <pathelement location="${javafx.graphics.jar}"/>
+        <pathelement location="/usr/share/java/javax.activation.jar"/>
+        <pathelement location="/usr/share/java/jaxb-api.jar"/>
         <!-- Not use at build time but used for unitary tests -->
         <pathelement location="${library.javasci.dir}/${library.javasci.name}"/>
         <pathelement location="${library.javasci.dir}/${library.javasci-v1.name}"/>
--- a/scilab/configure.ac
+++ b/scilab/configure.ac
@@ -1012,9 +1012,9 @@
                 AC_SUBST(SKINLF)
 
                 # Javax Annotation (Build dependency)
-                AC_JAVA_CHECK_JAR([javax.annotation-api],[javax.annotation.Generated],[Scilab Graphics - CSS Validation])
-                JAVAX_ANNOTATION=$PACKAGE_JAR_FILE
-                AC_SUBST(JAVAX_ANNOTATION)
+                #AC_JAVA_CHECK_JAR([javax.annotation-api],[javax.annotation.Generated],[Scilab Graphics - CSS Validation])
+                #JAVAX_ANNOTATION=$PACKAGE_JAR_FILE
+                #AC_SUBST(JAVAX_ANNOTATION)
 
                 # Jakarta Activation API (Build dependency)
                 AC_JAVA_CHECK_JAR([jakarta.activation-api],[jakarta.activation.spi.MimeTypeRegistryProvider],[Scilab Help - MimeTypes for image conversion])
--- a/scilab/modules/helptools/src/java/org/scilab/modules/helptools/java/JavaLexer.java
+++ b/scilab/modules/helptools/src/java/org/scilab/modules/helptools/java/JavaLexer.java
@@ -12,7 +12,6 @@
 
 import org.scilab.modules.helptools.c.AbstractCCodeHandler;
 
-@javax.annotation.Generated("JFlex")
 
 @SuppressWarnings("fallthrough")
 public final class JavaLexer {
